examples/gno.land/r/gnoland/boards2/v1/permissions_default.gno

```diff
@@ -0,0 +1,273 @@
+	})
+
+	return dp
+}
+
+// SetSuperRole assigns a super role.
+// A super role is one that have all permissions.
+// These type of role doesn't need to be mapped to any permission.
+func (dp *DefaultPermissions) SetSuperRole(r Role) {
+	dp.superRole = r
+}
+
+// AddRole add a role with one or more assigned permissions.
+func (dp *DefaultPermissions) AddRole(r Role, p Permission, extra ...Permission) {
+	dp.roles.Set(string(r), append([]Permission{p}, extra...))
+}
+
+// RoleExists checks if a role exists.
+func (dp DefaultPermissions) RoleExists(r Role) bool {
+	if dp.superRole != "" && r == dp.superRole {
+		return true
+	}
+
+	return dp.roles.Iterate("", "", func(name string, _ interface{}) bool {
+		return Role(name) == r
+	})
+}
+
+// GetUserRoles returns the list of roles assigned to a user.
+func (dp DefaultPermissions) GetUserRoles(user std.Address) []Role {
+	v, found := dp.users.Get(user.String())
+	if !found {
+		return nil
+	}
+	return v.([]Role)
+}
+
+// HasRole checks if a user has a specific role assigned.
+func (dp DefaultPermissions) HasRole(user std.Address, r Role) bool {
+	for _, role := range dp.GetUserRoles(user) {
+		if role == r {
+			return true
+		}
+	}
+	return false
+}
+
+// HasPermission checks if a user has a specific permission.
+func (dp DefaultPermissions) HasPermission(user std.Address, perm Permission) bool {
+	for _, r := range dp.GetUserRoles(user) {
+		if dp.superRole == r {
+			return true
+		}
+
+		v, found := dp.roles.Get(string(r))
+		if !found {
+			continue
+		}
+
+		for _, p := range v.([]Permission) {
+			if p == perm {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// AddUser adds a new user to permissions.
+func (dp *DefaultPermissions) AddUser(user std.Address, roles ...Role) error {
+	if dp.users.Has(user.String()) {
+		return errors.New("user already exists")
+	}
+
+	dp.dao.Members().Add(user)
+	return dp.setUserRoles(user, roles...)
+}
+
+// SetUserRoles sets the roles of a user.
+func (dp *DefaultPermissions) SetUserRoles(user std.Address, roles ...Role) error {
+	if !dp.users.Has(user.String()) {
+		return errors.New("user not found")
+	}
+	return dp.setUserRoles(user, roles...)
+}
+
+// RemoveUser removes a user from permissions.
+func (dp *DefaultPermissions) RemoveUser(user std.Address) bool {
+	_, removed := dp.users.Remove(user.String())
+	dp.dao.Members().Remove(user)
+	return removed
+}
+
+// HasUser checks if a user exists.
+func (dp DefaultPermissions) HasUser(user std.Address) bool {
+	return dp.users.Has(user.String())
+}
+
+// UsersCount returns the total number of users the permissioner contains.
+func (dp DefaultPermissions) UsersCount() int {
+	return dp.users.Size()
+}
+
+// IterateUsers iterates permissions' users.
+func (dp DefaultPermissions) IterateUsers(start, count int, fn UsersIterFn) bool {
+	return dp.users.IterateByOffset(start, count, func(k string, v interface{}) bool {
+		return fn(User{
+			Address: std.Address(k),
+			Roles:   v.([]Role),
+		})
+	})
```

Comment:
```suggestion
	return dp.dao.Members().IterateByOffset(start, count, func(k string, v interface{}) bool {
		addr := std.Address(k)
		roles := dp.GetUserRoles(addr)
		return fn(User{
			Address: addr,
			Roles:   roles,
		})
	})
```

Reply:
Addressed in [6f1205c](https://github.com/gnolang/gno/pull/2901/commits/6f1205c714b9072c2054f1c1167a573557e8a244)